packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -141,29 +212,46 @@ export class QueuedRequestController extends BaseController<
    * requests, ensuring they are executed one after the other to prevent concurrency issues and maintain proper
    * execution flow.
    *
-   * @param requestNext - A function representing the request to be enqueued. It returns a promise that
+   * @param request - The JSON-RPC request to process.
+   * @param requestNext - A function representing the next steps for processing this request. It returns a promise that
    * resolves when the request is complete.
    * @returns A promise that resolves when the enqueued request and any subsequent asynchronous
    * operations are fully processed. This allows you to await the completion of the enqueued request before continuing
    * with additional actions. If there are multiple enqueued requests, this function ensures they are processed in
    * the order they were enqueued, guaranteeing sequential execution.
    */
-  async enqueueRequest(requestNext: (...arg: unknown[]) => Promise<unknown>) {
+  async enqueueRequest(
```

Comment:
Previously, it seems like this method acted as a mutex, allowing one request to be filtered through the middleware chain at a time. (There is no queue defined here — it's really defined by the JavaScript engine.)

Perhaps we should embrace the generic nature of this method and call this `handleRequest`? So it waits until the current request is finished, or it processes it and then continues on to the next.